src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs

```diff
@@ -97,10 +97,16 @@ private void ExecuteSearchElement(ListBoxItem listBoxItem)
                 if (searchElement.CreateAndConnectCommand.CanExecute(port.PortModel))
                 {
                     searchElement.CreateAndConnectCommand.Execute(port.PortModel);
+                    var searchElementInfo = ViewModel.IsDisplayingMLRecommendation ?
+                        searchElement.FullName + " " + port.PortModel.Index.ToString() + " " + port.PortName + " " + port.NodeViewModel.OriginalName + " " +
```

Comment:
@QilongTang For a searchElement, we will have 2 conditions. If the searchElement is a zero-touch node, then we should use the searchElement.Model.CreationName and if the searchElement is a nodemodel node, then we should use the FullName,Assembly name like we serialize in DYN. Similarly for the target node, we should use port.NodeViewModel.NodeModel.CreationName in case of a zero-touch node.

Reply:
Hey @reddyashish How to check from searchElement if it is node model node? I debugged and see type is always zerotouch even for Color Palette?

Reply:
The NodeSearchElementViewModel should differ in its Model type. For zerotouch nodes, the searchElement.Model should be of type ZeroTouchSearchElement and for nodemodel nodes, the searchElement.Model should be of NodeModelSearchElement.

Reply:
Gotcha, just want to let you know, changing how this info is sent will also affect Object type way of AutoComplete, because in the past we have been only using `searchElement.FullName` for both cases.

Reply:
You mean it will affect the Analytics data and not the search results, right?